backend/hdf5/h5x/H5DataType.cpp

```diff
@@ -16,6 +16,11 @@ namespace nix {
 namespace hdf5 {
 namespace h5x {
 
+bool DataType::equal(const DataType &other) const {
+    HTri res = H5Tequal(hid, other.hid);
+    res.check("DataType::equal(): H5Tequal failed");
+    return res.result();
+}
```

Comment:
out of curiosity: why not the  ``==`` operator?

Reply:
Fair point. I guess I have been programming in plain C for too long. Added it.